src/app/posts/[id]/page.tsx

```diff
@@ -0,0 +1,152 @@
+"use client";
+import React, { useState } from "react";
+import { useParams } from "next/navigation";
+import { useEffect } from "react";
+import { useRouter } from "next/navigation";
+import { auth, db } from "@/firebase/config";
+import { getAuth, onAuthStateChanged, signOut } from "firebase/auth";
+import styles from "./page.module.css";
+import { useUserAuth } from "@/context/userAuth";
+import {
+  collection,
+  doc,
+  getDoc,
+  getDocs,
+  onSnapshot,
+  orderBy,
+  query,
+} from "firebase/firestore";
+import Sidebar from "@/compontents/Sidebar";
+import MiddleInput from "@/compontents/MiddleInput";
+import MiddlePost from "@/compontents/MiddlePost";
+import RightSideBar from "@/compontents/RightSideBar";
+import CommentModal from "@/compontents/CommentModal";
+import { RecoilRoot, useRecoilState } from "recoil";
+import { userState } from "@/../atom_state/userAtom";
+import Head from "next/head";
+import { FaArrowLeft } from "react-icons/fa6";
+import FirebasePost from "@/compontents/FirebasePost";
+import PostCompontent from "@/compontents/postPageCompontents/PostComponent";
+import UserComment from "@/compontents/postPageCompontents/UserComment";
+
+type NewsData = {
+  newsData: any[];
+};
+interface UserData {
+  email: string;
+  uid: string;
+  image: string;
+  text: string;
+  timestamp: string;
+  firstName: string;
+}
+
+const PostPage = ({ newsResults }: any) => {
+  const { user, setIsLoggedOut, setUser } = useUserAuth();
+
+  const router = useRouter();
+  const [currentUser, setCurrentUser] = useRecoilState(userState);
+
+  const idToString: any = useParams();
+  const id = idToString.id;
+  const [post, setPost] = useState<any>([]);
```

Comment:
@Kamildeeal you should define type for `Post` and `Comment` in `src/types` and then use it here